src/Middleware/HealthChecks/src/HealthCheckOptions.cs

```diff
@@ -28,7 +28,7 @@ public class HealthCheckOptions
         /// Gets a dictionary mapping the <see cref="HealthStatus"/> to an HTTP status code applied to the response.
         /// This property can be used to configure the status codes returned for each status.
         /// </summary>
-        public IDictionary<HealthStatus, int> ResultStatusCodes { get; } = new Dictionary<HealthStatus, int>()
+        public IDictionary<HealthStatus, int> ResultStatusCodes { get; set; } = new Dictionary<HealthStatus, int>()
```

Comment:
Please convert this into long form property (backing field with explict getter and setter) and add a null check for the setter. I don't want it to be possible to set this to null.

Reply:
I'll do it, but I’m currently stuck with #4605.